### PR TITLE
Refactor orgnaization policy management

### PR DESCRIPTION
**Pull request by Hinton**
## Objective
Refactors the policy management page for organizations. Instead of having a single component `PolicyEditComponent` containing a multitude of if statements, we now have a `BasePolicy` and `BasePolicyComponent` which encapsulates a single Policy. The `PoliciesComponents` displays a list of components which it retrives from the `PolicyListService`, and `PolicyEditComponent` lets the `BasePolicyComponent` handle the policy specific details.

I've also removed the deprecation warning for policies since it's no longer accurate.

### Noteworthy Code Changes
- **src/app/app.component.ts**: Policies are added during the bootstrap phase, not quite happy with the placement but I don't really see a good alternative.
- **src/app/organizations/manage/policy-edit.component.ts**: Removed policy specific logic since it now exists in separate components. To load the policy component we needed to use a `componentFactoryResolver`.

### Testing Considerations
Since this PR does some drastic changes to the policy management pages, we will have to do a full regression surrounding it.

Asana: https://app.asana.com/0/1198901840263430/1200796078487139

### CODE REVIEW

**Review by cscharf — COMMENTED** [review]
Oscar I think this looks great and is a good refactor for our policies pattern. I'll let one other person on the team take a look and see if I missed anything as far as implementation details or technicalities.

File: src/app/app.component.ts [hunk 1]
```diff
@@ -170,6 +182,18 @@ export class AppComponent implements OnDestroy, OnInit {
...
+            new MasterPasswordPolicy(),
+            new PasswordGeneratorPolicy(),
+            new SingleOrgPolicy(),
+            new RequireSsoPolicy(),
+            new PersonalOwnershipPolicy(),
+            new DisableSendPolicy(),
+            new SendOptionsPolicy(),
+            new ResetPasswordPolicy(),
+        ]);
+
```
Comment (addisonbeck) [inline src/app/app.component.ts:196]: I know you weren't happy about loading policies here, but I think it's perfect besides being in app.component. We get clear control and vision on what policies are supported & the addPolicies call taking an array is pretty clean.

Just curious, could we not populate the service in policies.component? What about in a constructor for the policy list service?

File: src/app/organizations/manage/policies.component.ts [hunk 2]
```diff
@@ -158,13 +102,11 @@ export class PoliciesComponent implements OnInit {
...
         });
-        this.policies.forEach(p => {
-            p.enabled = this.policiesEnabledMap.has(p.type) && this.policiesEnabledMap.get(p.type);
-        });
+
         this.loading = false;
     }
 
-    edit(p: any) {
+    edit(p: BasePolicy) {
```
Comment (addisonbeck) [inline src/app/organizations/manage/policies.component.ts:109]: I know you didn't name it, but I'd rather see a clearer name for this param.

```suggestion
    edit(policy: BasePolicy) {
```

File: src/app/organizations/manage/policies.component.ts [hunk 3]
```diff
@@ -174,9 +116,7 @@ export class PoliciesComponent implements OnInit {
         const childComponent = this.modal.show<PolicyEditComponent>(
             PolicyEditComponent, this.editModalRef);
 
-        childComponent.name = p.name;
-        childComponent.description = p.description;
-        childComponent.type = p.type;
+        childComponent.policy = p;
```
Comment (addisonbeck) [inline src/app/organizations/manage/policies.component.ts:119]:
```suggestion
        childComponent.policy = policy;
```

File: src/app/organizations/manage/policy-edit.component.html [hunk 4]
```diff
@@ -2,178 +2,21 @@
...
-                    </div>
+            
+            <div class="modal-body">
+                <div class="modal-body" *ngIf="loading">
+                    <i class="fa fa-spinner fa-spin text-muted" title="{{'loading' | i18n}}" aria-hidden="true"></i>
+                    <span class="sr-only">{{'loading' | i18n}}</span>
+                </div>
+                <div [hidden]="loading">
+                    <p>{{policy.description | i18n}}</p>
+                    <ng-template #policyForm></ng-template>
```
Comment (addisonbeck) [inline src/app/organizations/manage/policy-edit.component.html:18]: Hooray for organized html 🥳

**Review by addisonbeck — COMMENTED** [review]
Just one question and some minor feedback. Looks great, Oscar!

**Pull request merged**